thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java

```diff
@@ -657,6 +657,10 @@ void updateSubscriptionGroup(ThirdEyePrincipal user, long oldAlertConfigID, Stri
         }
       }
     }
+
+    if (!properties.containsKey(PROP_CLASS_NAME)) {
```

Comment:
The parsing of "DEFAULT_ALERTER_PIPELINE" type to the appropriate class happens in the translator. So you need to move this inside the translator class.

```
private Map<String, Object> buildAlerterProperties(Map<String, Object> alertYamlConfigs) {
    // By default we assume the subscription pipeline follows DEFAULT_ALERTER_PIPELINE
    alertYamlConfigs.putIfAbsent(PROP_TYPE, "DEFAULT_ALERTER_PIPELINE");  // ADD here
   ...
} 
```

Reply:
Makes sense! Moved the code.